params/config.go

```diff
@@ -444,6 +446,12 @@ func (c *ChainConfig) String() string {
 	if c.GrayGlacierBlock != nil {
 		banner += fmt.Sprintf(" - Gray Glacier:                %-8v (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/gray-glacier.md)\n", c.GrayGlacierBlock)
 	}
+	if c.ShangaiBlock != nil {
+		banner += fmt.Sprintf(" - Shangai:                     %-8v (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/shangai.md)\n", c.ShangaiBlock)
```

Comment:
https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/shangai.md -- this is a 404.